activities.py

```diff
@@ -0,0 +1,72 @@
+# @@@SNIPSTART python-money-transfer-project-template-withdraw
+from temporalio import activity
+
+from banking_service import BankingService, InvalidAccountError
+from shared import PaymentDetails
+
+
+# Generic errors for activity failures
+class WithdrawFailedError(Exception):
+    pass
+
+
+class DepositFailedError(Exception):
+    pass
+
+
+class RefundFailedError(Exception):
+    pass
+
+
+@activity.defn
+async def withdraw(data: PaymentDetails) -> str:
+    reference_id: str = f"{data.reference_id}-withdrawal"
+    bank = BankingService("bank-api.example.com")
```

Comment:
Usually this would be at the class level and each activity method would access the already-created service instead of connecting each time

Reply:
Made the Activity a Class.
Instantiated it in the Worker.